uitools/common/src/Internal/PopupAttachmentImageProvider.cpp

```diff
@@ -0,0 +1,198 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#include "PopupAttachmentImageProvider.h"
+
+// Qt headers
+#include <QAbstractFileIconProvider>
+#include <QFuture>
+#include <QIcon>
+#include <QMimeDatabase>
+
+// Maps SDK headers
+#include <Item.h>
+#include <PopupAttachment.h>
+
+// Toolkit headers
+#include "PopupAttachmentItem.h"
+
+namespace Esri::ArcGISRuntime::Toolkit {
+
+namespace {
+
+/*!
+  \internal
+  This class is an internal implementation detail and is subject to change.
+ */
+class PopupAttachmentImageResponse : public QQuickImageResponse
+{
+public:
+  PopupAttachmentImageResponse(PopupAttachmentItem* popupAttachmentItem, QSize requestedSize);
+  ~PopupAttachmentImageResponse() override;
+  QQuickTextureFactory* textureFactory() const override;
+
+private:
+  QPointer<PopupAttachmentItem> m_popupAttachmentItem;
+  QSize m_requestedSize;
+};
+
+PopupAttachmentImageResponse::PopupAttachmentImageResponse(PopupAttachmentItem* popupAttachmentItem, QSize requestedSize) :
+  m_popupAttachmentItem(popupAttachmentItem),
+  m_requestedSize(std::move(requestedSize))
+{
+  if (!m_popupAttachmentItem)
+  {
+    emit finished();
+    return;
+  }
+
+  auto thumbnail = popupAttachmentItem->thumbnail();
+  if (!thumbnail.isNull())
+  {
+    // We have a thumbnail to display so just go ahead and display this.
+    emit finished();
+    return;
+  }
+
+  auto item = m_popupAttachmentItem;
+  if (!item)
+  {
+    emit finished();
+    return;
+  }
+
+  auto itemThumbnail = item->thumbnail();
+  if (!itemThumbnail.isNull())
+  {
+    // We have a good thumbnail.
+    emit finished();
+    return;
+  }
+
+  auto popupAttachment = item->popupAttachment();
+  if (popupAttachment->popupAttachmentType() != PopupAttachmentType::Image)
+  {
+    emit finished();
+    return;
+  }
+}
+
+PopupAttachmentImageResponse::~PopupAttachmentImageResponse() = default;
+
+QQuickTextureFactory* PopupAttachmentImageResponse::textureFactory() const
+{
+  QImage thumbnail;
+  QIcon icon;
+
+  if (m_popupAttachmentItem)
+  {
+    thumbnail = m_popupAttachmentItem->thumbnail();
+  }
+
+  // Handles edge case for Mobile since QAbstractFileIconProvider returns a Null QIcon.
+#if defined(Q_OS_IOS) || defined(Q_OS_ANDROID)
+  if (thumbnail.isNull() || m_popupAttachmentItem->popupAttachmentType() != PopupAttachmentType::Image)
+  {
+    switch (m_popupAttachmentItem->popupAttachmentType()) {
+      case PopupAttachmentType::Image:
+        icon = QIcon(":/esri.com/imports/Calcite/images/image.svg");
+        break;
+      case PopupAttachmentType::Video:
+        icon = QIcon(":/esri.com/imports/Calcite/images/video.svg");
```

Comment:
assuming these work, even without qrc path defined? sometimes they get tripped up on mobile if not referenced inside of the qrc

Reply:
They have been working but i'll put it in just in case.

Reply:
Ah this doesn't work on mobile with the `qrc` passed into the QIcon constructor. I was following BasemapGalleryImageProvider for this, https://github.com/Esri/arcgis-maps-sdk-toolkit-qt/blob/2193418c264fb8e841d017b753f84c4aa131b572/uitools/common/src/Internal/BasemapGalleryImageProvider.cpp#L135. I am guessing it's because of how QIcon looks up the path.https://doc.qt.io/qt-6/qicon.html#QIcon-5